### PR TITLE
Reach volume step by step

### PR DESCRIPTION
**Pull request by 2t0m**
Hi @crowbarz,

As you may know, several Pioneer AVR don't support the `set_volume_level` function but only `volume_up` and `volume_down`.
In order to be able de control volume with the slider, I suggest to add a 2nd volume option I called 'volume_steps' and use a loop to reach the volume target with `volume_up` and `volume_down`.

Let me know what you think.

### CODE REVIEW

File: custom_components/pioneer_async/pioneer_avr.py [hunk 1]
```diff
@@ -860,11 +862,20 @@ async def set_volume_level(self, volume, zone="1"):
...
         ):
             raise ValueError(f"volume {volume} out of range for zone {zone}")
-        vol_len = 3 if zone == "1" else 2
-        vol_prefix = str(volume).zfill(vol_len)
-        return await self.send_command(
-            "set_volume_level", zone, prefix=vol_prefix, ignore_error=False
-        )
+        if self.volume_steps:
+            current_volume = self.volume.get(zone)
+            steps = abs(round((volume - current_volume) / 2))
```
Comment (crowbarz) [inline custom_components/pioneer_async/pioneer_avr.py:867]: `volume` is measured in the relevant units for the zone (0.5db increments for Zone 1, 1db increments for other zones.) So should this be dividing the volume difference by 2 to determine how many commands to send? On my AVR, the volume up/down commands skip by 0.5db on Zone 1 and 1db on HDZone.

File: custom_components/pioneer_async/pioneer_avr.py [hunk 2]
```diff
@@ -860,11 +862,20 @@ async def set_volume_level(self, volume, zone="1"):
...
+            for x in range(steps):
+                if volume > current_volume:
+                    await self.volume_up()
```
Comment (crowbarz) [inline custom_components/pioneer_async/pioneer_avr.py:870]: should be `await self.volume_up(zone=zone)` to work with zones other than 1, same for volume_down.

**Comment by crowbarz** [comment]
Just to let you know, I'm in the middle of splitting `pioneer_avr.py` into a separate repository [crowbarz/aiopioneer](http://github.com/crowbarz/aiopioneer), so I'll try to merge this branch into that work later on.

**Review by crowbarz — CHANGES_REQUESTED** [review]

File: custom_components/pioneer_async/pioneer_avr.py (hunk 1 above)
Comment (2t0m) [reply custom_components/pioneer_async/pioneer_avr.py:867]: My VSX-S510 works differently. The volume can be adjusted from 0 to 151 volume units and the volume up/down commands skip by 2 volume units.

Your function `query_device_info()` works well for me.

Do you think adding an exception is a good idea ?

```
if self.model == "VSX-S510":
        self.volume_steps = True
	self.max_volume["1"] = 151
	self.volume_steps_units = [2]
---

if self.volume_steps:
	current_volume = self.volume.get(zone)
	steps = abs(round((volume - current_volume) / self.volume_steps_units[int(zone)]))
		for x in range(steps):
	if volume > current_volume:
		await self.volume_up()
	elif volume < current_volume:
		await self.volume_down()
```

Or group AVR in families is better ?
Normal Family (VSX-5XX, VSX-8XX, VSX-9XX) and Slim Family (VSX-SXXX) ?

File: custom_components/pioneer_async/pioneer_avr.py (hunk 2 above)
Comment (2t0m) [reply custom_components/pioneer_async/pioneer_avr.py:870]: Understood

**Review by 2t0m — COMMENTED** [review]
Do you think it's a good idea to group AVR models in families (normal and slim) ?

File: custom_components/pioneer_async/pioneer_avr.py (hunk 1 above)
Comment (crowbarz) [reply custom_components/pioneer_async/pioneer_avr.py:867]: I am not a great fan of hard-coded exceptions :) However, for now, let's leave it as a separate option that supports VSX-S510 specifically.

Going forward, I would like `PioneerAVR` to be able to determine appropriate values for currently hard-coded defaults for an AVR (eg. min/max volume, volume up/down step) based on model number, as well as HA being able to query those defaults (to use as defaults in the UI) as well as set them if the user chooses to do so via either the UI or `configuration.yaml`. I'd rather complete the separation of the API currently in progress before embarking on this, so I've created crowbarz/aiopioneer#1 to track that work.

**Comment by crowbarz** [comment]
The options framework I'm planning will determine appropriate defaults based on regex match on device model, so arbitrary groupings should be possible (eg. match on `^VSX-S` or `^VSX-9[234]0` `^VSX-S510$).

**Review by crowbarz — APPROVED** [review]

**Comment by crowbarz** [comment]
FYI I renamed `*_VOLUME_STEPS_ONLY` to `*_VOLUME_STEPS` to be more consistent with the other options, and also reworded the option label to make it clear it's VSX-S510 specific.

**Comment by 2t0m** [comment]
Good luck for the remaining work. Let me know if I can provide specific information about the model I own.

**Pull request merged**